### PR TITLE
New integration options for D&D

### PR DESCRIPTION
**Pull request by jpople**
Closes PROD-2155

### Description Of Changes

Adds UI to configure integrations and add discovery monitors for S3, Scylla, and DynamoDB to admin UI.

Screenshot 2024-06-18 at 14 51 32
Screenshot 2024-06-18 at 14 51 41
Screenshot 2024-06-18 at 14 51 56
Screenshot 2024-06-18 at 14 52 17

(Note these screenshots don't show the correct logo for S3, but I've fixed that!)

### Code Changes

* [ ] Refactors to new integrations UI to handle arbitrary integration types rather than specifically BigQuery
* [ ] Changes new integration detail view to force redirect if an integration not supported for D&D is opened
* [ ] Adds tab-based filtering to integrations index and integration type selection screen
* [ ] Refactors "show more/less" button in copy to ShowMoreContent component
* [ ] Adds new FidesSpinner component to ensure spinners are centered in their containers rather than top-left-aligned
* [ ] Re-generates OpenAPI types

### Steps to Confirm

In new integrations UI:
* [ ] Create a Scylla, DynamoDB, and S3 integration and add a discovery monitor for each
* [ ] View integrations of those types on the list view and each of their detail pages
* [ ] Forms should be populated with appropriate parameters per integration type and appropriate copy (lorem ipsum for now)

### Pre-Merge Checklist

* [x] Issue Requirements are Met
* [X] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

### CODE REVIEW

File: clients/admin-ui/src/features/connection-type/connection-type.slice.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ import {
 import type { RootState } from "../../app/store";
 import {
   ConnectionTypeParams,
-  ConnectionTypeSecretSchemaReponse,
+  ConnectionTypeSecretSchemaResponse,
```
Comment (jpople) [inline clients/admin-ui/src/features/connection-type/connection-type.slice.ts:17]: This was just a typo that got preserved from way back when.

File: clients/admin-ui/src/features/integrations/useIntegrationFilterTabs.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  const [tabIndex, setTabIndex] = useState(0);
+  const currentTab = tabs[tabIndex];
+
+  const [isFiltering, setIsFiltering] = useState(false);
+
+  const onChangeFilter = (newIndex: number) => {
+    setIsFiltering(true);
+    setTabIndex(newIndex);
+    setTimeout(() => setIsFiltering(false), 100);
```
Comment (jpople) [inline clients/admin-ui/src/features/integrations/useIntegrationFilterTabs.tsx:22]: Putting in a fake "loading" state for a short period makes the change in contents feel less jarring to me-- not married to this if we're against it.

File: clients/admin-ui/src/features/common/FidesSpinner.tsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { Flex, Spinner, SpinnerProps } from "fidesui";
+
+const FidesSpinner = ({ ...props }: SpinnerProps) => (
```
Comment (lucanovera) [inline clients/admin-ui/src/features/common/FidesSpinner.tsx:3]: good one making this reusable

File: clients/admin-ui/src/features/common/copy/ShowMoreContent.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { Collapse, Text, useDisclosure } from "fidesui";
+import { ReactNode } from "react";
+
+const ShowMoreContent = ({ children }: { children: ReactNode }) => {
```
Comment (lucanovera) [inline clients/admin-ui/src/features/common/copy/ShowMoreContent.tsx:4]: yes, reusability!

File: clients/admin-ui/src/features/integrations/ConfigureIntegrationModal.tsx [hunk 5]
```diff
@@ -1,7 +1,8 @@
 import { UseDisclosureReturn } from "fidesui";
 
 import AddModal from "~/features/configure-consent/AddModal";
-import ConfigureIntegrationForm from "~/features/integrations/ConfigureIntegrationForm";
+import ConfigureIntegrationForm from "~/features/integrations/add-integration/ConfigureIntegrationForm";
+import useIntegrationOption from "~/features/integrations/useIntegrationOption";
 import { ConnectionConfigurationResponse } from "~/types/api";
 
 const ConfigureIntegrationModal = ({
```
Comment (lucanovera) [inline clients/admin-ui/src/features/integrations/ConfigureIntegrationModal.tsx:8]: composition for the win!

File: clients/admin-ui/src/features/integrations/integration-type-info/s3Info.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  created_at: "",
+};
+
+export const S3_TAGS = ["Data Warehouse", "S3", "Tag 3", "Tag 4"];
+
+export const S3Overview = () => (
+  <>
+    <InfoHeading text="Overview" />
+    <InfoText>
+      Here&apos;s some example copy talking about Amazon S3. Lorem ipsum dolor
```
Comment (lucanovera) [inline clients/admin-ui/src/features/integrations/integration-type-info/s3Info.tsx:26]: Are the real copies ready?

File: clients/admin-ui/src/features/integrations/integration-type-info/dynamoInfo.tsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  created_at: "",
+};
+
+export const DYNAMO_TAGS = ["Database", "DynamoDB", "Tag 3", "Tag 4"];
+
+export const DynamoOverview = () => (
+  <>
+    <InfoHeading text="Overview" />
+    <InfoText>
+      Here&apos;s some example copy talking about DynamoDB. Lorem ipsum dolor
```
Comment (lucanovera) [inline clients/admin-ui/src/features/integrations/integration-type-info/dynamoInfo.tsx:26]: Do we have the final copy?

File: clients/admin-ui/src/features/integrations/useIntegrationFilterTabs.tsx (hunk 2 above)
Comment (lucanovera) [reply clients/admin-ui/src/features/integrations/useIntegrationFilterTabs.tsx:22]: I think small artificial delays are valid if it improves the UX. We can keep it.

**Review by lucanovera — CHANGES_REQUESTED** [review]
Overall the code looks very good! Great work. Just a question about the missing copies for some integrations. 

And two UI bugs:
1. This cancel button seems to be much bigger than all of the other ones, I think it should be the same size
Captura de pantalla 2024-06-19 a la(s) 4 15 04 p  m

2. The active tab highlight. I think there may be a separate issue for this, but if there's a quick fix for it, it would be nice to add it. It's very explicit because when you open the modal the first tab already has the highlight rectangle.
Captura de pantalla 2024-06-19 a la(s) 4 14 30 p  m

**Review by lucanovera — APPROVED** [review]
Thanks for the button fix. approve

**Pull request merged**